Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -475,143 +469,201 @@ namespace OpenInfraPlatform {
+				// SUBTYPE OF(IfcRectangleProfileDef);
+				// END_ENTITY;
+				// **************************************************************************************************************************
+				double x = rounded_rectangle->XDim * UnitConvert()->getLengthInMeterFactor();
+				double y = rounded_rectangle->YDim * UnitConvert()->getLengthInMeterFactor();
+				std::vector<carve::geom::vector<2>> outer_loop;
+				if (rounded_rectangle->RoundingRadius) {
+					AddRectangleCoordinates(outer_loop, rounded_rectangle->RoundingRadius * UnitConvert()->getLengthInMeterFactor(), x, y);
 					paths.push_back(outer_loop);
-					return;
 				}
+				return;
+			}
+			/*! \brief defines a standard rectangle
+			*  \param[in] rectangle_profile A pointer to data from \c IfcRectangleProfileDef
+			*  \param[out] paths A pointer to be filled with the relevant data
+			*/
+			void convertIfcStandardRectangleProfileDef(const EXPRESSReference<typename IfcEntityTypesT::IfcRectangleProfileDef>& rectangle_profile, 
+				std::vector<std::vector<carve::geom::vector<2>>>& paths) const {
+				// **************************************************************************************************************************
+				// ENTITY IfcStandardRectangleProfileDef
+				// -
+				// SUBTYPE OF(IfcRectangleProfileDef);
+				// END_ENTITY;
+				// **************************************************************************************************************************
+				double x = rectangle_profile->XDim * UnitConvert()->getLengthInMeterFactor();
+				double y = rectangle_profile->YDim * UnitConvert()->getLengthInMeterFactor();
+				std::vector<carve::geom::vector<2>> outer_loop;
+				AddRectangleCoordinates(outer_loop, 0.0, x, y);
+				paths.push_back(outer_loop);
+				return;
 			}
 
-			// Function 6.2  convertIfcTrapeziumProfileDef SUBTYPE OF IfcParametrizedProfileDef
+			/*! \brief defines a trapezium as the profile definition
+			*  \param[in] trapezium A pointer to data from \c IfcParametrizedProfileDef
+			*  \param[out] paths A pointer to be filled with the relevant data
+			*/
 			void convertIfcTrapeziumProfileDef(const EXPRESSReference<typename IfcEntityTypesT::IfcTrapeziumProfileDef>& trapezium,
-				std::vector<std::vector<carve::geom::vector<2>>>& paths, std::vector<carve::geom::vector<2>> outer_loop) const {
+				std::vector<std::vector<carve::geom::vector<2>>>& paths) const {
+				// **************************************************************************************************************************
+				// ENTITY IfcTrapeziumProfileDef
+				// https://standards.buildingsmart.org/IFC/DEV/IFC4_2/FINAL/HTML/schema/ifcprofileresource/lexical/ifctrapeziumprofiledef.htm
+				// SUBTYPE OF(IfcParametrizedProfileDef);
+				// END_ENTITY;
+				// **************************************************************************************************************************

```

Comment:
The ENTITY definitions are not correct. Please see the Schema file and include the correct definitions.

Reply:
resolved:  8838897